gcp_variant_transforms/libs/bq_to_vcf_temp_files_composer.py

```diff
@@ -0,0 +1,144 @@
+
+  It composes VCF data shards to one VCF data file and deletes the intermediate
+  VCF shards.
+  TODO(allieychen): Eventually, it further consolidates the meta information,
+  data header line, and the composed VCF data file into the `output_file`.
+
+  Args:
+    project: The project name.
+    output_file: The final VCF file path.
+    bq_to_vcf_temp_folder: The folder that contains all BigQuery to VCF pipeline
+     temp files.
+  """
+  bucket_name, blob_prefix = gcsio.parse_gcs_path(bq_to_vcf_temp_folder)
+  client = storage.Client(project)
+  bucket = client.get_bucket(bucket_name)
+  multi_process_composer = MultiProcessComposer(project, bucket_name,
+                                                blob_prefix)
+  composed_vcf_data_file = multi_process_composer.run()
+  output_file_blob = _create_blob(client, output_file)
+  output_file_blob.rewrite(composed_vcf_data_file)
+  bucket.delete_blobs(bucket.list_blobs(prefix=blob_prefix))
+
+
+def _compose_files(project, bucket_name, blob_names, composite_name):
+  # type: (str, str, List[str], str) -> None
+  """Composes multiple files (up to 32 objects) in GCS to one.
+
+  Args:
+    project: The project name.
+    bucket_name: The name of the bucket where the `components` and the new
+      composite are saved.
+    blob_names: A list of blob object names.
+    composite_name: Name of the new composite.
+  """
+  bucket = storage.Client(project).get_bucket(bucket_name)
+  output_file_blob = bucket.blob(composite_name)
+  output_file_blob.content_type = 'text/plain'
+  blobs = [bucket.get_blob(blob_name) for blob_name in blob_names]
+  output_file_blob.compose(blobs)
+
+
+def _create_blob(client, file_path):
+  # type: (storage.Client, str) -> storage.Blob
+  bucket_name, blob_name = gcsio.parse_gcs_path(file_path)
+  file_blob = client.get_bucket(bucket_name).blob(blob_name)
+  file_blob.content_type = 'text/plain'
+  return file_blob
+
+
+class MultiProcessComposer(object):
+  """Class to compose (a large number of) files in GCS in parallel."""
+
+  _NUM_PROCS = 63
+
+  def __init__(self, project, bucket_name, blob_prefix):
+    # type: (str, str, str) -> None
+    """Initialize a `MultiProcessComposer`.
+
+    This class composes all blobs that start with `blob_prefix` to one.
+
+    Args:
+      project: The project name.
+      bucket_name: The name of the bucket where the blob components and the new
+        composite are saved.
+      blob_prefix: The prefix used to filter blobs. Only the VCF files with this
+        prefix will be composed.
+    """
+    self._project = project
+    self._bucket_name = bucket_name
+    self._blob_prefix = blob_prefix
+    self._bucket = storage.Client(project).get_bucket(bucket_name)
+
+  def run(self):
+    # type: () -> storage.Blob
+    """Returns the final blob that all blobs composed to."""
+    return self._compose_vcf_shards_to_one(self._blob_prefix)
+
+  def _compose_vcf_shards_to_one(self, blob_prefix):
```

Comment:
nit: since this logic is not VCF-specific, consider removing VCF from the name and comments. I think `_compose_blobs_to_one` may be a better name?